BankManagerConsoleApp/BankManagerConsoleApp/Model/Bank.swift

```diff
@@ -1,40 +1,66 @@
 import Foundation
 
 final class Bank {
-    private let bankClerk: BankClerk
-    private var customerQueue = Queue<Customer>()
+    private var loanCustomerQueue = Queue<Customer>()
+    private var depositCustomerQueue = Queue<Customer>()
+    private let loanBankersNumber: Int
+    private let depositBankersNumber: Int
+    private var numberOfCustomers = 0
     
-    init(bankClerk: BankClerk) {
-        self.bankClerk = bankClerk
+    init(loanBankersCount: Int = 1, depositBankersCount: Int = 2) {
+        self.loanBankersNumber = loanBankersCount
+        self.depositBankersNumber = depositBankersCount
     }
     
     func handOutWaitingNumber(from customerNumber: Int) {
         for number in 1...customerNumber {
             let customer = Customer(waitingNumber: number)
-            customerQueue.enqueue(customer)
+            customerQueue(customer.banking).enqueue(customer)
         }
     }
     
-    private func respondToCustomer() {
-        guard let customer = customerQueue.dequeue() else {
-            return
+    private func customerQueue(_ banking: Banking) -> Queue<Customer> {
+        switch banking {
+        case .loan:
+            return loanCustomerQueue
+        case .deposit:
+            return depositCustomerQueue
         }
-        bankClerk.work(for: customer)
     }
     
     func openBank() {
-        var numberOfCustomers = 0
+        let bankGroup = DispatchGroup()
         let openTime = Date()
-        while customerQueue.isEmpty == false {
-            respondToCustomer()
-            numberOfCustomers += 1
-        }
+        
+        workBankers(loanBankersNumber, customers: loanCustomerQueue, group: bankGroup)
+        workBankers(depositBankersNumber, customers: depositCustomerQueue, group: bankGroup)
+        
+        bankGroup.wait()
         closeBank(numberOfCustomers, from: openTime)
     }
     
+    private func workBankers(_ number: Int, customers: Queue<Customer>, group: DispatchGroup) {
+        for _ in 1...number {
+            workBanker(customers: customers, group: group)
+        }
+    }
+    
+    private func workBanker(customers: Queue<Customer>, group: DispatchGroup) {
+        let banker = Banker()
+        DispatchQueue.global().async(group: group) {
+            while customers.isEmpty == false {
+                guard let customer = customers.dequeue() else {
+                    fatalError()
+                }
+                self.numberOfCustomers += 1
+                banker.work(for: customer)
+            }
+        }
+    }
```

Comment:
```suggestion
   private func workBanker(customers: Queue<Customer>, group: DispatchGroup) {
        let banker = Banker()
        DispatchQueue.global().async(group: group) {
            while let customer = customers.dequeue() {
                banker.work(for: customer)
            }
        }
    }
```

들여쓰기 vs 가독성
관련해서는 위처럼 코드를 개선하는 방향으로도 수정할 수 있을 것 같아요!
 메서드가 가능한 하나의 일만 하도록 잘 쪼개져 있는지 판단하는게 먼저일 것 같구요. 그게 어느정도 잘 되어있고 (중복을 줄이고 네이밍을 신경쓰는 등의) 코드 개선이 어느정도 된 상태라면  두가지 다 크게 문제될 건 없을 것 같습니다.

Reply:
오.. 왜 while let이 생각이 안났나 모르겠어요. 😂
피드백 반영해서 코드 수정 해보았습니다. 좋은 의견 주셔서 감사합니다!!

Reply:
델마! 이번에도 좋은 피드백 감사드립니다! 🤩 👍 

아래와 같이 수정해보았습니다. 
코멘트로 질문도 남겨드렸는데 확인 부탁드립니다~ 🙏🏻

## 수정사항

- Bank 타입 리팩토링
    - 은행원 수를 나타내는 프로퍼티명 number를 count로  수정
    - Bank타입을 class가 아닌 struct로 수정
    - `workBanker()` 메소드 내부에 while문을 while let으로 수정
- Banker 타입의 `workSpeed()` 메소드를 제거
- LinketList의 `removeFirst()` 메소드 내부를 Thread-Safe하게 개선
    - Thread Safe
        - 기존 로직은 은행원 수만큼 global() 큐에 Task를 만드는 방식이여서 하나의 고객 큐에 여러 스레드가 접근해 `Race Condition`이 발생할 가능성이 있음.
        - 여러 스레드가 하나의 고객 큐에 접근해도 LinkedList에서 첫 번째 요소를 반환하는 메서드를 하나의 스레드에서만 접근할 수 있도록 `SerialQueue.sync`를 활용하여 `Race Condition`을 해결

코멘트 확인해주시고 문제없다면 merge 부탁드립니다!
감사합니다~ 😊